accepted/0000-workspaces-auto-cwd-context.md

```diff
@@ -0,0 +1,73 @@
+# npm workspaces: auto switch context based on cwd
+
+## Summary
+
+Running **workspace-aware** commands from the context of a child **workspace** folder should work as expected.
+
+## Motivation
+
+Most users expect to be able to navigate to a **nested workspace** directory and use npm commands in that context.
+
+Unfortunately that might not work as intended depending on the command, for example a user trying to add a new dependency `<pkg>` would expect to `cd ./<workspace-folder>` and `npm install <pkg>` to produce a valid install tree adding the new dependency to that workspace but instead that creates a separate install within that directory, failing to hoist/dedupe dependencies to the top-level and generating an extra and unintended package-lock file.
+
+## Detailed Explanation
+
+Now that we are landing support to **workspace-aware** commands (see: [#117](https://github.com/npm/rfcs/pull/117)) a potential fix to that mismatch between the expected user experience and the current behavior of the npm cli is to automatically switch the internal context to ensure that if the user is trying to run a command that already support the **workspaces configurations**.
+
+## Rationale and Alternatives
+
+- Use the logic to figuring out if a **workspace-aware** command was run from the context of a child **workspace** folder but only print a warning message to users, similar to our existing set of "Did you mean?" warning messages. Educating the user on the usage of the `--workspace` config rather than automatically switching the context.
+- Only implement the auto-switch behavior in arborist-reify related methods. namely: `ci|dedupe|find-dupes|install-ci-test|install-test|install|link|rebuild|update|uninstall|update`
+- Make it so that the auto-switch behavior of the cli is disabled by default and needs some type of opt-in mechanism.
+- Implement the auto-switch behavior as an opt-in with very explicit ways of activating it (e.g: via a config option that can be set in `.npmrc` or cli)
+- Do not implement any auto-switch mechanism
+
+## Implementation
+
+Add a recursive check that walks up the file system structure and validates if the current `prefix` is defined as the **"workspaces"** property of a `package.json` file of a parent directory as part of `@npmcli/config`.
```

Comment:
Please do not use the "prefix" config for anything new. It breaks nvm, and the `$PREFIX` env var breaks termux.

Reply:
oh don't worry this is only internal implementation details here but now that you mentioned I double checked and it's actually incorrect, it should mention `localPrefix` instead

Reply:
ref: https://github.com/npm/config/pull/28

Reply:
thanks for confirming :-)